helix-term/src/ui/editor.rs

```diff
@@ -684,6 +685,14 @@ impl EditorView {
                 });
             let text = Text::styled(&diagnostic.message, style);
             lines.extend(text.lines);
+            let code = diagnostic.code.as_ref().map(|x| match x {
+                NumberOrString::Number(n) => format!("({n})"),
+                NumberOrString::String(s) => format!("({s})"),
+            });
+            if let Some(code) = code {
+                let text = Text::styled(code, style);
+                lines.extend(text.lines);
```

Comment:
Using `Text::styled` is a little bit overkill here. It's used in the above block because the diagnostic message may have newlines in it but we know for sure that this span will not. We can use `Span` instead which won't try to parse any lines or look at the text:

```suggestion
                let span = Span::styled(code, style);
                lines.push(span.into());
```